src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -122,14 +127,42 @@ public boolean isRunning() {
      * Runs verifyIotCertificate useCase for all the registered client certificate PEMs.
      */
     @Override
-    public void run() {
+    public synchronized void run() {
         if (isNetworkDown()) {
             logger.debug("Network is down - not refreshing certificates");
             return;
         }
 
+        if (!canRun()) {
+            return;
+        }
+
         Optional<ThingAssociations> associations = getThingsAssociatedWithCoreDevice();
-        associations.ifPresent(this::refresh);
+        associations.ifPresent(a -> {
+            this.refresh(a);
+            nextRun.set(Instant.now().plus(Duration.ofSeconds(DEFAULT_INTERVAL_SECONDS)));
```

Comment:
Now this may potentially not refresh for close to 48 hours. The next scheduled refresh needs to be pushed when this happens, right?